### DISCUSSION TITLE
Is it possible to make eslint not ignore inner node_modules ?

### DISCUSSION THREAD

**Discussion by axellbrendow** [eslint/eslint#14681] [Q&A] [answered]
I have a project structured like this:

```
.eslintignore
.eslintrc.json
node_modules // packages from npm install
src
    node_modules // Layers of my project
        controllers
        services
        repositories
```

I could not get ESLint to not ignore the inner node_modules.

Inside .eslintignore I have `!src/node_modules/**/*.js` and I tried also running `npx eslint --no-ignore --fix .`

I expected that ESLint would let me tell it to not ignore that node_modules.

The intention with the inner node_modules is that I can import my controllers, services and repositories without using `../../../../../../`

**Comment by nzakas** [comment]
The ignore system is a bit hairy, but you should be able to unignore inner node_modules. Try running with —debug to see what’s going on.

**Comment by mdjermanovic** [comment, marked as answer]
ESLint has a predefined list of patterns, with one element:

```
/**/node_modules/*
```

Patterns from `.eslintignore` are appended to this list, so at the end you have:

```
/**/node_modules/*
!src/node_modules/**/*.js
```

This list still matches _directories_ like `src/node_modules/controllers`, and per [the .gitignore specification](https://git-scm.com/docs/gitignore) it is not possible to re-include a file if a parent directory of that file is excluded.

Can you try with the following pattern in `.eslintignore`:

```
!/src/node_modules/*
```

and then run ESLint:

```
npx eslint --fix .
```

(without `--no-ignore`, as it tells ESLint to not load the `.eslintignore` file, but the predefined pattern still applies)

**Reply by axellbrendow** [reply]
I tested now and it worked! Thank you so much!

Thanks for mentioning the .gitignore specification too!
